### PR TITLE
cli(mode): set mode to none if none is set after defaults

### LINKED ISSUE #407: Warning message says --mode=none is supported, but using --mode=none produces error
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

Running without `--mode` results in a warning that includes a message about using `--mode=none`:

```sh
$ webpack
...
WARNING in configuration
The 'mode' option has not been set, webpack will fallback to 'production' for this value. Set 'mode' option to 'development' or 'production' to enable defaults for each environment.
You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/concepts/mode/
```

Yet running with `--mode=none` produces the `--help` text, followed by this message:

```sh
$ webpack --mode=none
...
Invalid values:
  Argument: mode, Given: "none", Choices: "development", "production"
```

**If the current behavior is a bug, please provide the steps to reproduce.**

``` sh
$ npm init --yes
$ npm i -D webpack webpack-cli
$ mkdir src
$ touch src/index.js
$ webpack
$ webpack --mode=none
```

**What is the expected behavior?**

Either:

1. Do not have a message claiming support for `--mode=none` or
2. Add support for `--mode=none`

**Please paste the results of `webpack-cli info` here, and mention other relevant information such as programming language.**

```
$ webpack-cli info

  System:
    OS: Windows 10
    CPU: x64 Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
  Binaries:
    Node: 9.11.1
    Yarn: 1.5.1
    npm: 5.8.0
  Browsers: {}
  npmPackages:
    webpack:
      wanted: ^4.6.0
      installed: 4.6.0
    webpack-cli:
      wanted: ^2.0.14
      installed: 2.0.14
```

### PR DESCRIPTION
**Pull request by ev1stensberg**
**What kind of change does this PR introduce?**

Bugfix

**Did you add tests for your changes?**
Y
**If relevant, did you update the documentation?**
Y
**Summary**
Sets mode to `none` if no mode is set after all mode options has been set.

N/A

**Does this PR introduce a breaking change?**
No

**Other information**
Fixes #407

### CODE REVIEW

**Comment by ev1stensberg** [comment]
@rossipedia https://github.com/webpack/webpack-cli/blob/master/bin/config-yargs.js#L90-L95 also has this, new version hasn't been published yet.

File: bin/convert-argv.js [hunk 1]
```diff
@@ -40,7 +40,9 @@ module.exports = function(...args) {
 		argv["output-filename"] = path.basename(output);
 		argv["output-path"] = path.dirname(output);
 	}
-
+	if(!argv.mode) {
```
Comment (PlayMa256) [inline bin/convert-argv.js]: can't this be set as `default: 'none'` in the [options](https://github.com/webpack/webpack-cli/blob/master/bin/config-yargs.js#L90-L95)? Same effect.

Comment (ev1stensberg) [reply bin/convert-argv.js]: The two statements above are relying on it not to be set, so that's why it's placed here

Comment (PlayMa256) [reply bin/convert-argv.js]: The same does not happen when you set `default: none` via `yargs`? If not set, it gets the default value? (just curiosity btw)

Comment (dhruvdutt) [reply bin/convert-argv.js]: Yup. I think @PlayMa256 is right. Config phase happens first.

Comment (ev1stensberg) [reply bin/convert-argv.js]: Okay gotcha, will rebase the tomorrow, internet here is pretty bad

Comment (ev1stensberg) [reply bin/convert-argv.js]: Actually I don't think we should set default as none, webpack will warn us, that's intuitive and it informs the user about the choices they're making

**Pull request closed without merging**